wandb/sdk/internal/internal_api.py

```diff
@@ -737,6 +737,87 @@ def upsert_project(self, project, id=None, description=None, entity=None):
         # TODO(jhr): Commenting out 'repo' field for cling, add back
         #   'description': description, 'repo': self.git.remote_url, 'id': id})
         return response["upsertModel"]["model"]
+    
+    @normalize_exceptions
+    def get_project_run_queues(self, entity, project):
+        query = gql("""
+        query Project($entity: String!, $projectName: String!){
+            project(entityName: $entity, name: $projectName) {
+                runQueues {
+                    id
+                    name
+                    createdBy
+                    access
+                }
+            }
+        }
+        """
+        )
+        variable_values = {
+            "projectName": project,
+            "entity": entity,
+        }
+        return self.gql(query, variable_values)["project"]["runQueues"]          # todo handle nonexistent queue exception 
+
+    @normalize_exceptions
+    def create_run_queue(self, entity, project, queue_name, access):
+        query = gql("""
+        mutation createRunQueue($entity: String!, $project: String!, $queueName: String!, $access: RunQueueAccessType!){
+            createRunQueue(
+                input: {
+                    entityName: $entity,
+                    projectName: $project,
+                    queueName: $queueName,
+                    access: $access
+                }
+            ) {
+                success
+                queueID
+            }
+            
+        }
+        """
+        )
+        variable_values = {
+            "project": project,
+            "entity": entity,
+            "access": access,
+            "queueName": queue_name,
+        }
+        return self.gql(query, variable_values)["createRunQueue"]
+
+    @normalize_exceptions
+    def push_to_run_queue(self, queue_name, run_spec):
+        # todo: we're adding pushToRunQueueByName to avoid this extra query
```

Comment:
Forgot about this TODO. https://wandb.atlassian.net/browse/WB-5798 If you want it, but I don't think we need to rush that